homeassistant/components/bosch_shc/switch.py

```diff
@@ -0,0 +1,219 @@
+                description=SWITCH_TYPES["smartplugcompact"],
+            )
+        )
+
+    for switch in session.device_helper.camera_eyes:
+
+        entities.append(
+            SHCCameraSwitch(
+                device=switch,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+                description=SWITCH_TYPES["cameraeyes"],
+            )
+        )
+
+    for switch in session.device_helper.camera_360:
+
+        entities.append(
+            SHCCameraSwitch(
+                device=switch,
+                parent_id=session.information.unique_id,
+                entry_id=config_entry.entry_id,
+                description=SWITCH_TYPES["camera360"],
+            )
+        )
+
+    if entities:
+        async_add_entities(entities)
+
+
+class SHCSwitch(SHCEntity, SwitchEntity):
+    """Representation of a SHC switch."""
+
+    def __init__(
+        self,
+        device: SHCDevice,
+        parent_id: str,
+        entry_id: str,
+        description: SHCSwitchEntityDescription,
+    ) -> None:
+        """Initialize a SHC switch."""
+        super().__init__(device, parent_id, entry_id)
+        self.entity_description = description
+
+    @property
+    def is_on(self):
+        """Return the state of the switch."""
+        return (
+            getattr(self._device, self.entity_description.on_key)
+            == self.entity_description.on_value
+        )
+
+    @property
+    def today_energy_kwh(self):
+        """Return the total energy usage in kWh."""
+        if self.entity_description.has_consumption:
+            return self._device.energyconsumption / 1000.0
+        return None
+
+    @property
+    def current_power_w(self):
+        """Return the current power usage in W."""
+        if self.entity_description.has_consumption:
+            return self._device.powerconsumption
+        return None
+
+    def turn_on(self, **kwargs):
+        """Turn the switch on."""
+        setattr(self._device, self.entity_description.on_key, True)
+
+    def turn_off(self, **kwargs):
+        """Turn the switch off."""
+        setattr(self._device, self.entity_description.on_key, False)
+
+    def toggle(self, **kwargs):
+        """Toggle the switch."""
+        setattr(self._device, self.entity_description.on_key, not self.is_on)
+
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        if self.entity_description.key == "smartplug":
+            return {
+                "routing": self._device.routing.name,
+            }
+        if self.entity_description.key == "smartplugcompact":
+            return {
+                "communication_quality": self._device.communicationquality.name,
+            }
+        return None
+
+
+class SHCCameraSwitch(SHCSwitch):
+    """Representation of a SHC camera switch."""
+
+    @property
+    def should_poll(self):
+        """Camera needs polling."""
+        return True
```

Comment:
Can you create a `should_poll` property on the entity_description?
```suggestion
        return self.entity_description.should_poll
```